lib/src/find_pattern.dart

```diff
@@ -0,0 +1,95 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// find_pattern.dart
+// Implementation of Find Pattern Functionality.
+//
+// 2025 March 07
+// Author: Ramli, Nurul Izziany <nurul.izziany.ramli@intel.com>
+//
+
+import 'package:rohd/rohd.dart';
+
+/// [FindPattern] functionality
+///
+/// Takes in a [Logic] to find location of a fixed-width pattern.
+/// Outputs pin `index` contains position.
+class FindPattern extends Module {
+  /// [index] is a getter for output of FindPattern
+  Logic get index => output('index');
```

Comment:
What does the index output when there is an error?  Would be good to document (even if you claim "undefined behavior")